internal/controller/controllers/agent_controller.go

```diff
@@ -1022,23 +1026,29 @@ func (r *AgentReconciler) updateLabels(log logrus.FieldLogger, ctx context.Conte
 	changed = setAgentLabel(log, agent, AgentLabelClusterDeploymentNamespace, namespace) || changed
 
 	if changed {
-		if err := r.Update(ctx, agent); err != nil {
-			log.WithError(err).Errorf("failed to add labels to agent %s/%s", agent.Namespace, agent.Name)
-			return err
-		}
-		agentRef := types.NamespacedName{Namespace: agent.Namespace, Name: agent.Name}
-		err := r.Get(ctx, agentRef, agent)
-		if err != nil {
-			log.WithError(err).Errorf("Failed to get agent %s", agentRef)
-			return err
-		}
+		return r.updateAndReplaceAgent(ctx, agent)
 	}
 
 	return nil
 }
 
+func (r *AgentReconciler) updateAndReplaceAgent(ctx context.Context, agent *aiv1beta1.Agent) error {
+	if err := r.Update(ctx, agent); err != nil {
+		return errors.Wrapf(err, "failed to update agent %s/%s", agent.Namespace, agent.Name)
+	}
+	agentRef := types.NamespacedName{Namespace: agent.Namespace, Name: agent.Name}
+	err := r.Get(ctx, agentRef, agent)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get agent %s", agentRef)
+	}
+	return nil
+}
+
 func setAgentAnnotation(log logrus.FieldLogger, agent *aiv1beta1.Agent, key string, value string) bool {
 	annotations := agent.GetAnnotations()
+	if annotations == nil {
```

Comment:
Is there a specific scenario it can be nil? Worth adding a test for it?

Reply:
I ran into this when running tests. I don't think it should really ever be nil in a real scenario.

I don't remember which of these tests I changed ran into it though. I could add an explicit one if you think it's needed.

Reply:
Oh ok, I was just curious when it happens:) No need for test if you don't think it's necessary.